django/db/models/functions.py

```diff
@@ -182,13 +180,11 @@ def __init__(self, expression, pos, length=None, **extra):
             expressions.append(length)
         super(Substr, self).__init__(*expressions, **extra)
 
-    def as_sqlite(self, compiler, connection):
-        self.function = 'SUBSTR'
-        return super(Substr, self).as_sql(compiler, connection)
+    def as_sqlite(self, compiler, connection, **kwargs):
```

Comment:
Agreed. Either pass `**kwargs` through to the underlying `as_sql` method, or remove them from the signature of `as_sqlite`. I'd lean towards passing kwargs through to remain consistent with the API. I can't think of any reason why the `as_sqlite` would ever need to accept `kwargs` (since it is the one doing the overriding) though. Tim?